Riot/Modules/SideMenu/SideMenuCoordinator.swift

```diff
@@ -220,6 +231,14 @@ final class SideMenuCoordinator: SideMenuCoordinatorType {
         self.spaceMenuPresenter.present(forSpaceWithId: spaceId, from: self.sideMenuViewController, sourceView: sourceView, session: session, animated: true)
     }
     
+    func navigate(to item: SpaceExploreRoomListItemViewData, from sourceView: UIView?) {
+        if item.childInfo.roomType == .space {
+            self.exploreRoomCoordinator?.pushSpace(with: item)
```

Comment:
To handle Coordinator navigation you should either manage navigation inside the `start()` method like in `KeyBackupRecoverCoordinator.start()` for a flow or in your case for a simple reusable screen you can handle the navigation from the outside like that:
```
    func createExploreRoomsCoordinator() -> ExploreRoomCoordinator {
        let exploreRoomCoordinator = ExploreRoomCoordinator(session: self.session, spaceId: self.spaceId)
        exploreRoomCoordinator.delegate = self
        let presentable = exploreRoomCoordinator.toPresentable()
    }

    func presentExploreRooms() {
        let presentingViewController = ....
        let exploreRoomCoordinator = self.createExploreRoomsCoordinator()
        let presentable = exploreRoomCoordinator.toPresentable()
        presentable.presentationController?.delegate = self
        presentingViewController.present(presentable, animated: animated, completion: nil)
        exploreRoomCoordinator.start()
        
        self.add(childCoordinator: exploreRoomCoordinator) // do not forget to remove the coordinator on dismiss
    }

    func pushExploreRooms() {
        let exploreRoomCoordinator = self.createExploreRoomsCoordinator()
        self.navigationRouter.push(exploreRoomCoordinator, animated: true) {
            self.remove(childCoordinator: coordinator)
        }
        exploreRoomCoordinator.start()
        
        self.add(childCoordinator: exploreRoomCoordinator)
    }
```

Reply:
I ported the navigation to the `ExploreRoomCoordinator`